### ISSUE TITLE
BUG: No taskbar icon

### ISSUE THREAD

**Issue by v4u6h4n** [StreamController/StreamController#87]
I guess this is a feature request if there isn't actually supposed to be one.

**Comment by Core447** [comment]
This has been requested by many people. I would like to have an icon myself, but I haven't come up with a good way to add one yet, because GTK4 has stopped providing an interface for them.

**Comment by RileyMeta** [comment]
So, GTK itself may have stopped providing support for it, but plugins like `KStatusNotifierItem` ~~`TopIcons Plus`~~ `Top Panel Notification Icons Revived` `Tray Icons: Reloaded` and `App Icons Taskbar` do bring it back and allow for Taskbar icons. 

Ex: 
![image](https://github.com/StreamController/StreamController/assets/32332593/2725dfd9-e87b-4300-9c1e-20dc0b879e48)

I think you could do something like back-porting GTK 3.0 and the app indicator support:
```py
import gi
gi.require_version('Gtk', '3.0')
from gi.repository import Gtk

def create_status_indicator():
    status_indicator = Gtk.StatusIcon()
    status_indicator.set_from_icon_name("icon-name")  # Replace "icon-name" with your icon
    status_indicator.set_title("StreamController")
    status_indicator.set_tooltip_text("An elegant Linux app for the Elgato Stream Deck with support for plugins")
    return status_indicator
```
You would need to modify the primary `__init__` and the function call that starts it, but IMHO it would be fairly stupid to mix versions. Alternatively you could try to add the QT Status Indicator support because those are supported by the above plugins. 

```py
from PyQt5.QtGui import QIcon

def create_status_indicator():
    status_indicator = QSystemTrayIcon()
    status_indicator.setIcon(QIcon("icon-path"))  # Replace "icon-path" with the path to your icon
    status_indicator.setToolTip("An elegant Linux app for the Elgato Stream Deck with support for plugins")
    return status_indicator
```

Very quickly scanning another project that has a working App Indicator, the only line of code I can see that links to the icon is 
```py
APP_LOGO = os.path.join(PROJECT_PATH, "logo.svg")`
# and
logo = QIcon(APP_LOGO)
```

> Although it does look like you might need to add [gdk_set_allowed_backends ("x11");](https://discourse.gnome.org/t/how-to-add-custom-icons-to-gnome-shell-tray-in-wayland-environment/5049/3#:~:text=Force%20the%20X11%20backend%20in%20the%20client%3A) to "backport" X11 icon support to Wayland. 

> [!NOTE]
> I am running Gnome 46.1 on X11, the plugin says it's currently out of date (Gnome 45) but works fine.
> (Granted I've only started App Development recently so I may get things wrong)

**Comment by Core447** [comment]
> Alternatively you could try to add the QT Status Indicator support because those are supported by the above plugins.

Even if I would, I couldn't use Gtk 3 and 4 in the same app, and using qt just for the indicator is a bit overkill (would need to add qt and the python package to the Flatpak).

**Comment by RileyMeta** [comment]
Understood.
I find it fairly unfortunate that they removed support for Taskbar icons completely in GTK 4, but oh well.
What about `libappindicator` or `statusnotifier`, could you do something like a single-file that just manages the Status Icon and have it load along-side the primary application?

**Comment by Core447** [comment]
Yes, I though about that, too. While it is not very nice, it might be our only option.

**Comment by RileyMeta** [comment]
I wonder if it would be better to basically write it like a small compatibility layer, where the GTK 3 Status Icon app starts first and runs a few quick checks to make sure everything is working (if Udev rules aren't present, then add them, etc). That way it could also issue commands directly to the main app; if the user were to right-click the icon it has a few options like: Opening the Configuration menu, or Closing the app Completely. 

Found an [explicit example](https://gist.github.com/sarnold/6fadb4b0dc438956c01fb83f6d664c8e) that might be able to be modified too

If it's kept as it's own single-file app then it could also help with containing any issues and trouble-shooting, along with allow users to customize it by replacing the icon to something that matches their theme (without knowledge of Python).

**Comment by Core447** [comment]
I wanted to add a small script that handles the real launching anyway that can notify users if there are any major bugs (that prevent the app from launching). However, I think I would also use GTK4 for this, so we might have to do something like this

```
Startup program (checking for bugs) ----> Main app
                                     |
                                     | -> App indicator
```

**Comment by Core447** [comment]
> This has been requested by many people. I would like to have an icon myself, but I haven't come up with a good way to add one yet, because GTK4 has stopped providing an interface for them.

Gonna close this because of this
